src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -78,16 +78,17 @@ ChipDeviceScanner::~ChipDeviceScanner()
 {
     StopScan();
 
-    // mTimerExpired should only be set to true in the TimerExpiredCallback, which means we are in that callback
-    // right now so there is no need to cancel the timer. Doing so would result in deadlock trying to aquire the
-    // chip stack lock which we already currently have.
-    if (!mTimerExpired)
-    {
+#if CHIP_STACK_LOCK_TRACKING_ENABLED
+    if (!DeviceLayer::PlatformMgr().IsChipStackLockedByCurrentThread())
```

Comment:
At the moment I feel like this is an anti pattern. Hence why this PR is in draft form currently.

But this is just taking the recommendation as is over here https://github.com/project-chip/connectedhomeip/commit/da37e592c3f3a2df3d4f1a2aaf13ac68e1e3e2fc#r112086198